### DISCUSSION TITLE
Prop type for drawerContent in a nested drawer navigator

### DISCUSSION THREAD

**Discussion by a-c-sreedhar-reddy** [react-navigation/react-navigation#9135] [Q&A]
My navigigation is 
```
Stack
  Drawer
    DrawerScreen1
    DrawerScreen2
  StackScreen

```

The `Drawer.Navigator` accepts a prop `drawerContent`. 

`drawerContent` accepts a function of type `(props) => React.Element`.

The type of `props.navigator` in `drawerContent` function is `DrawerNavigationHelpers`. This does not allow `navigation.push`. 

How can we assign a type so that `props.navigator` has both `drawer and stack` methods?

StackOverflow : https://stackoverflow.com/questions/65214148/prop-type-for-drawercontent-in-a-nested-drawer-navigator

**Comment by satya164** [comment]
You'll need to cast the type if you want it to be something other than `DrawerNavigationHelpers`. Like

```js
(navigation as StackNavigationProp).push(Whatever)
```

**Comment by a-c-sreedhar-reddy** [comment]
@satya164 
```
Conversion of type 'DrawerNavigationProp<RootDrawerParamList, string>' to type 'StackNavigationProp<StackParamList, string>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```
so I had to do this 
```ts
 ((navigation as any) as StackNavigationProp<StackParamList>).push( 'Videos');
```
But I will be losing the type-safety right?
I could not understand https://reactnavigation.org/docs/typescript/#nesting-navigators this clearly. Would it help?
